helm-chart/openj9-jitserver-chart/README.md

```diff
@@ -0,0 +1,228 @@
+``` yaml
+image:
+  repository: adoptopenjdk
+  tag: 8-openj9
+  pullPolicy: Always
+  command: ["jitserver"]
+```
+
+Should you want to deploy AdoptOpenJDK Java 11 images, change `image` fields inside `Values.yaml` as shown below.
+``` yaml
+image:
+  repository: adoptopenjdk
+  tag: 11-openj9
+  pullPolicy: Always
+  command: ["jitserver"]
+```
+
+The complete list of avaliable images and tags can be found at [AdoptOpenJDK Docker Hub](https://hub.docker.com/_/adoptopenjdk?tab=tags).
+
+### JITServer and application deployment topology considerations
+One or more application(s) can connect to one JITServer instance as long as they are running the same JDK version and enough resources are allocated for JITServer. However, a single application should not be allowed to connect to multiple JITServer instances. 
+
+JITServer can be deployed either on the same node or on a different node from the java application. There are benefits from both methods.
+
+* Deploy JITServer on the same node as the java application
+If the node that java application runs on has enough resources, deploying JITServer on the same node is recommended since this reduces networking overheads and maximizes JITServer performance. Note that this may not always be possible, if you want to connect multiple JVM clients to a single JITServer instance.
+
+* Deploy JITServer on a different node as the java application
+If the node that java application runs on does not have enough CPU or memory resources, JITServer can be deployed on a different node. 
+
+You may specify the node to deploy on by editing `affinity.nodeAffinity.requiredDuringSchedulingIgnoredDuringExecution[].values[]` in `Values.yaml`. Node names can be found by `kubectl get nodes`.
+``` yaml
+affinity: 
+  nodeAffinity:
+    requiredDuringSchedulingIgnoredDuringExecution:
+        - key: kubernetes.io/hostname
+          operator: In
+          values:
+          - "example-node-name"
+```
+
+## Application and JITServer life cycle management
+
+### Verifying if the JITServer is deployed
+``` bash
+$ helm status jitserver-release
+NAME: jitserver-release
+LAST DEPLOYED: Wed Nov 18 18:41:07 2020
+NAMESPACE: helm-chart-test
+STATUS: deployed
+REVISION: 1
+NOTES:
+Welcome to OpenJ9 JITServer Helm Chart, the application has been deployed successfully.
+```
+
+``` bash
+$ helm test jitserver-release
+Pod jitserver-release-openj9-jitserver-chart-test-connection pending
+Pod jitserver-release-openj9-jitserver-chart-test-connection succeeded
+NAME: jitserver-release
+LAST DEPLOYED: Wed Nov 18 19:19:46 2020
+NAMESPACE: helm-chart-test
+STATUS: deployed
+REVISION: 1
+TEST SUITE:     jitserver-release-openj9-jitserver-chart-test-connection
+Last Started:   Wed Nov 18 19:41:14 2020
+Last Completed: Wed Nov 18 19:41:26 2020
+Phase:          Succeeded
+NOTES:
+Welcome to OpenJ9 JITServer Helm Chart, the application has been deployed successfully.
+```
```

Comment:
Is there a way to make this report the `java -version` / `jitserver -version` from the deployed jitserver?  It would help users to diagnose mismatches between the jdk for the jitserver and their application

Reply:
Since this is a plant text txt file, normal scripts cannot be executed while being printed to users. I have added two commands that users can run on their environment to obtain `java -version` of the container. 

The `helm install` command prints below message. 

```
$ helm install openj9 openj9-jitserver-chart/
NAME: openj9
LAST DEPLOYED: Sun Jan  3 19:12:00 2021
NAMESPACE: jitserver
STATUS: deployed
REVISION: 1
NOTES:
Welcome to OpenJ9 JITServer Helm Chart. The JITServer technology has been deployed successfully and ready for connection.

1. Printing version of the chart and application
  Chart version: 0.23.0
  OpenJ9 release: 0.23.0

2. Get Java version by running these commands:
  export POD_NAME=$(kubectl get pod --namespace jitserver -o jsonpath="{..metadata.name}" | grep openj9-openj9-jitserver-chart)
  kubectl exec $POD_NAME -i -t -- java -version

3. Get JITServer endpoint by running these commands:
  export SERVICE_IP=$(kubectl get svc --namespace jitserver openj9-openj9-jitserver-chart -o jsonpath="{..metadata.name}")
  echo Endpoint: $SERVICE_IP:38400
```